modules/develop/pages/data-transforms/run-transforms.adoc

```diff
@@ -0,0 +1,254 @@
++
+The following examples show some basic transforms. Each example can be copied into the `transform.go` file.
++
+[tabs]
+====
+Identity transform::
++
+--
+```go
+package main
+
+import (
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows the basic usage of the package:
+// This transform does nothing but copy the same data from an
+// input topic to an output topic.
+func main() {
+	// Make sure to register your callback and perform other setup in main
+	redpanda.OnRecordWritten(identityTransform)
+}
+
+// This will be called for each record in the source topic.
+//
+// The output records returned will be written to the destination topic.
+func identityTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	return []redpanda.Record{e.Record()}, nil
+}
+```
+--
+Transcoder transform::
++
+--
+
+```go
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"io"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows a transform that converts CSV inputs into JSON outputs.
+func main() {
+	redpanda.OnRecordWritten(csvToJsonTransform)
+}
+
+type Foo struct {
+	A string `json:"a"`
+	B int    `json:"b"`
+}
+
+func csvToJsonTransform(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	// The input data is a CSV (without a header row) that is the structure of:
+	// key, a, b
+	reader := csv.NewReader(bytes.NewReader(e.Record().Value))
+	// Improve performance by reusing the result slice.
+	reader.ReuseRecord = true
+	output := []redpanda.Record{}
+	for {
+		row, err := reader.Read()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, err
+		}
+		if len(row) != 3 {
+			return nil, errors.New("unexpected number of rows")
+		}
+		// Convert the last column into an int
+		b, err := strconv.Atoi(row[2])
+		if err != nil {
+			return nil, err
+		}
+		// Marshal our JSON value
+		f := Foo{
+			A: row[1],
+			B: b,
+		}
+		v, err := json.Marshal(&f)
+		if err != nil {
+			return nil, err
+		}
+		// Add our output record using the first column as the key.
+		output = append(output, redpanda.Record{
+			Key:   []byte(row[0]),
+			Value: v,
+		})
+
+	}
+	return output, nil
+}
+```
+
+--
+Validation filter transform::
++
+--
+```go
+import (
+	"encoding/json"
+
+	"github.com/redpanda-data/redpanda/src/go/transform-sdk"
+)
+
+// This example shows a filter that outputs only valid JSON into the
+// output topic.
+func main() {
+	redpanda.OnRecordWritten(filterValidJson)
+}
+
+func filterValidJson(e redpanda.WriteEvent) ([]redpanda.Record, error) {
+	v := []redpanda.Record{}
+	if json.Valid(e.Record().Value) {
+		v = append(v, e.Record())
+	}
+	return v, nil
+}
+```
+
+--
+====
+
+== Build and deploy the transform
+
+. Build the transform into a Wasm module with metadata: 
++
+```bash
+rpk transform build
+```
+
+. Deploy the Wasm module to your cluster. For example:
```

Comment:
@micheleRP  this doesn't make it clear we're deploying the identity transform (one of the 3 above). let's state that ... so the output topic messages are expected to be identical.  

Or, to make the point more illustrative, show the CSV--> JSON  transform input.output.